### PR TITLE
perf(db): improved performance of disable/delete of an entity

### PR DESCRIPTION
**Pull request by jeabakker**
fixes: #10798

### CODE REVIEW

File: engine/classes/ElggEntity.php [hunk 1]
```diff
@@ -1797,22 +1797,28 @@ public function disable($reason = "", $recursive = true) {
...
-			]);
-			$subentities->setIncrementOffset(false);
+				'limit' => false,
+			];
 			
-			foreach ($subentities as $subentity) {
-				/* @var $subentity \ElggEntity */
-				if (!$subentity->isEnabled()) {
-					continue;
+			foreach (['owner_guid', 'container_guid', 'site_guid'] as $db_column) {
```
Comment (hypeJunction) [inline engine/classes/ElggEntity.php]: Might save cycles if you reverse the array order

Comment (jeabakker) [inline engine/classes/ElggEntity.php]: i don't think so, i think more entities are owned by an entity than they have site_guid as that entity.

Comment (jdalsem) [inline engine/classes/ElggEntity.php]: it will never matter... entities will only appear once... it does not matter in which result set

**Review by hypeJunction — APPROVED** [review]

**Review by jdalsem — APPROVED** [review]

**Comment by hypeJunction** [comment]
This broke the build in master

**Comment by jdalsem** [comment]
create a new issue

**Comment by hypeJunction** [comment]
I am commenting here because it hasn't been tested properly before merging. There is an infinite loop somewhere.

**Pull request merged**